x-pack/plugins/security_solution/server/lib/telemetry/tasks/prebuilt_rule_alerts.ts

```diff
@@ -47,53 +50,62 @@ export function createTelemetryPrebuiltRuleAlertsTaskConfig(maxTelemetryBatch: n
           return 0;
         }
 
-        const processedAlerts = telemetryEvents.map(
-          (event: TelemetryEvent): TelemetryEvent =>
-            copyAllowlistedFields(filterList.prebuiltRulesAlerts, event)
-        );
-
-        const sanitizedAlerts = processedAlerts.map(
-          (event: TelemetryEvent): TelemetryEvent =>
-            processK8sUsernames(clusterInfo?.cluster_uuid, event)
-        );
-
-        const enrichedAlerts = sanitizedAlerts.map(
-          (event: TelemetryEvent): TelemetryEvent => ({
-            ...event,
-            licence_id: licenseInfo?.uid,
-            cluster_uuid: clusterInfo?.cluster_uuid,
-            cluster_name: clusterInfo?.cluster_name,
-            package_version: packageInfo?.version,
-          })
-        );
-
-        tlog(logger, `sending ${enrichedAlerts.length} elastic prebuilt alerts`);
-        const batches = batchTelemetryRecords(enrichedAlerts, maxTelemetryBatch);
-        for (const batch of batches) {
-          await sender.sendOnDemand(TELEMETRY_CHANNEL_DETECTION_ALERTS, batch);
+        let fetchMore = true;
+        let searchAfterValue: SortResults | undefined;
+        let pitId = await receiver.openPointInTime(index);
+
+        while (fetchMore) {
+          const { moreToFetch, newPitId, searchAfter, alerts } =
+            await receiver.fetchPrebuiltRuleAlertsBatch(pitId, searchAfterValue);
+
+          if (alerts.length === 0) {
+            return 0;
+          }
+
+          fetchMore = moreToFetch;
+          searchAfterValue = searchAfter;
+          pitId = newPitId;
+
+          const processedAlerts = alerts.map(
+            (event: TelemetryEvent): TelemetryEvent =>
+              copyAllowlistedFields(filterList.prebuiltRulesAlerts, event)
+          );
+
+          const sanitizedAlerts = processedAlerts.map(
+            (event: TelemetryEvent): TelemetryEvent =>
+              processK8sUsernames(clusterInfo?.cluster_uuid, event)
+          );
+
+          const enrichedAlerts = sanitizedAlerts.map(
+            (event: TelemetryEvent): TelemetryEvent => ({
+              ...event,
+              licence_id: licenseInfo?.uid,
+              cluster_uuid: clusterInfo?.cluster_uuid,
+              cluster_name: clusterInfo?.cluster_name,
+              package_version: packageInfo?.version,
+              task_version: taskVersion,
+            })
+          );
+
+          tlog(logger, `sending ${enrichedAlerts.length} elastic prebuilt alerts`);
+          const batches = batchTelemetryRecords(enrichedAlerts, maxTelemetryBatch);
+
+          const promises = batches.map(async (batch) => {
+            sender.sendOnDemand(TELEMETRY_CHANNEL_DETECTION_ALERTS, batch);
+          });
+
+          await Promise.all(promises);
         }
-        await sender.sendOnDemand(TASK_METRICS_CHANNEL, [
-          createTaskMetric(taskName, true, startTime),
-        ]);
-        return enrichedAlerts.length;
+
+        await receiver.closePointInTime(pitId);
```

Comment:
I think this code won't be executed when any promises in `Promise.all` (L104) fail, but it should, right?